airflow/utils/entry_points.py

```diff
@@ -18,13 +18,14 @@
 
 import functools
 import logging
+import sys
 from collections import defaultdict
 from typing import Iterator, Tuple
 
-try:
-    import importlib_metadata as metadata
-except ImportError:
-    from importlib import metadata  # type: ignore[no-redef]
+if sys.version_info >= (3, 12):
+    from importlib import metadata
+else:
+    import importlib_metadata as metadata  # type: ignore[no-redef]
```

Comment:
To explain this-one:

From: https://docs.python.org/3/library/importlib.metadata.html#entry-points

> Compatibility Note

> The “selectable” entry points were introduced in importlib_metadata 3.6 and Python 3.10. Prior to those changes, entry_points accepted no parameters and always returned a dictionary of entry points, keyed by group. With importlib_metadata 5.0 and Python 3.12, entry_points always returns an EntryPoints object. See [backports.entry_points_selectable](https://pypi.org/project/backports.entry-points-selectable) for compatibility options.